### DISCUSSION TITLE
Issues with the comprehensive test

### DISCUSSION THREAD

**Discussion by kmayne01** [neuropoly/axondeepseg#723] [Help needed] [answered]
Hello,

I have successfully installed axondeepseg on my windows computer but now have access to a Linux system where I am having issues with the installation. It passed the quick test but I am getting error codes when trying to run the comprehensive test (after just copying and pasting the script from the guidelines):

py.test: error: unrecognized arguments: --cov --cov-report term-missing

Any help would be appreciated!

**Comment by mathieuboudreau** [comment, marked as answer]
Hi @kmayne01,

`--cov` is for the "code coverage analysis", it's mainly used by us developers. Although this in itself isn't a problem (techniquely you could simply try running `pytest .` instead), it should have been installed during the conda environment step (specifically [here](https://github.com/axondeepseg/axondeepseg/blob/f5044aa1edebdf9b2f954785606963cb0ea34764/environment.yml#L22)), so the error is likely an indication that either 1) your conda virtual environment isn't properly "activated" or 2) the conda environment installation didn't work as expected.

If you'd like us to further help you debug that issue, you could try running the following commands in your terminal and sharing the output for each.

* `conda list`
* `pip freeze`
* `echo $CONDA_PREFIX`
* `which pytest`

**Reply by kmayne01** [reply]
@mathieuboudreau so I started over completely and it has worked fine this time! Not sure what I did differently but thank you for your suggestions and help.

**Reply by mathieuboudreau** [reply]
@kmayne01 Great to hear!

From the message that you sent earlier (presumably that you deleted, but I got in my emails), it does in fact appear that at the very least, the ADS venv wasn't activated. This is because the output of the `echo $CONDA_PREFIX`,

/home/ninc-user/miniconda/envs/env

didn't point to a virtual env but actually the "base" miniconda env, and that `which pytest` 

/usr/bin/pytest

was pointing to a system-wide pytest and not a miniconda or even ADS virtualenv location.

Trying the installation procedures from scratch is what I would have recommended, so I'm glad that worked out for you!
